Git/config.sublime-syntax

```diff
@@ -0,0 +1,206 @@
+%YAML 1.2
+---
+# Syntax based on documentation here:
+# https://git-scm.com/docs/git-config#_syntax
+
+name: Git Config
+file_extensions:
+  - gitconfig                # /etc/gitconfig
+  - .gitconfig               # ~/.gitconfig
+first_line_match: ^\[core\]  # .git/config files always start with [core]
+scope: source.gitconfig
+
+variables:
+  escape_char: \\"bnt
+  section_name: '[\w.-]+(?=[\s\]])'  # lookahead catches missing space, e.g. color"branch".
+  space_or_eol: (?=(?:\s|$))         # makes values on last line match properly
+  variable_name: '[a-zA-Z][\w-]*'
+  zero_to_255: 25[0-5]|2[0-4][0-9]|1\d\d|[1-9][0-9]|[0-9]
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: section
+
+  boolean:
+    # 0 and 1 are omitted due to mismatches (e.g. "log -1", "HEAD~1"). They will
+    # be matched as unquoted strings instead.
+    - match: \b(?:yes|true|on|off|no|false){{space_or_eol}}
+      scope: constant.language.gitconfig
+
+  color:
+    - include: color-named      # red, blue, green
+    - include: color-attribute  # bold, italic, underline
+    - include: color-numeric    # 0-255
+
+  color-attribute:
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink){{space_or_eol}}
+      scope: support.constant.color-attribute.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  color-numeric:
+    - match: '\b(?:{{zero_to_255}}){{space_or_eol}}'
+      scope: constant.other.color.rgb-value.gitconfig
+
+  comments:
+    - match: '[#;]'
+      scope: punctuation.definition.comment.gitconfig
+      push:
+        - meta_scope: comment.line.gitconfig
+        - match: $\n?  # consume newline so completions aren't shown at EOL when
+          pop: true    # typing comments
+
+  # The only valid escapes: '\b', '\n', '\t', '\"', '\\'.
+  escape:
+    - include: trailing-slash
+    - match: \\[^{{escape_char}}]
+      scope: invalid.illegal.escape.gitconfig
+    - match: \\[{{escape_char}}]
+      scope: constant.character.escape.gitconfig
+
+  illegal-newline:
+    - match: \n
+      scope: invalid.illegal.gitconfig
+
+  illegal-slash:
+    - match: \\
+      scope: invalid.illegal.gitconfig
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
```

Comment:
By convention this needs to be: `meta.mapping.gitconfig` where `{{variable_name}}` is a `meta.mapping.key` and all the values are scoped as `meta.mapping.value`.

You can have a look into the JSON syntax or https://github.com/SublimeText/PackageDev/tree/master/Package files, which handle it very well.

Reply:
The JSON syntax doesn't handle this yet, but you can check out the pull request for it here: https://github.com/sublimehq/Packages/pull/862